turbopack/crates/turbo-tasks-backend/src/backend/operation/update_collectible.rs

```diff
@@ -32,26 +32,24 @@ impl UpdateCollectibleOperation {
         let outdated = get!(task, OutdatedCollectible { collectible }).copied();
         if let Some(outdated) = outdated {
             if count > 0 && outdated > 0 {
-                update_count!(
-                    task,
-                    OutdatedCollectible { collectible },
-                    -min(count, outdated)
-                );
+                let shared = min(count, outdated);
+                update_count!(task, OutdatedCollectible { collectible }, -shared);
+                count -= shared;
             } else if count < 0 && outdated < 0 {
-                update_count!(
-                    task,
-                    OutdatedCollectible { collectible },
-                    min(-count, -outdated)
-                );
+                let shared = min(-count, -outdated);
```

Comment:
I'm not sure what a negative value means for `count` or `outdated` here, but do you want to use `min` in both cases? Should one of these be `max`?

Reply:
When you `take` collectibles, this is implemented as emitting negative collectibles for the current task. So the negative collectibles negate the positive collections and they are effectively gone.

min is correct here as the `-` sign is inside of the arguments. It could also be `-max(count, outdated)`, which would be equal. I find it easier to think about min/max with positive number so I converted the negative numbers to positive one before applying `min`